deltachat-ios/Coordinator/AppCoordinator.swift

```diff
@@ -100,11 +100,12 @@ class AppCoordinator: NSObject, Coordinator {
 
     func showChat(chatId: Int) {
         showTab(index: 3)
-        let navController = self.chatListController as! UINavigationController
-        let chatVC = ChatViewController(dcContext: dcContext, chatId: chatId)
-        let coordinator = ChatViewCoordinator(dcContext: dcContext, navigationController: navController, chatId: chatId)
-        chatVC.coordinator = coordinator
-        navController.pushViewController(chatVC, animated: true)
+        if let navController = self.chatListController as? UINavigationController {
```

Comment:
@r10s here it might be better to use a `guard let` instead of `if let` in combination with an `assertionFailure` (this fails in debug mode) - because it would be a programming error / something would be seriously wrong if the `chatListController` would not cast to a `UINavigationController` here:

    guard let navController = self.chatListController as? UINavigationController else {
        assertionFailure("huh? why no nav controller?")
    }

And the guard let makes the control flow more clear IMO.
See https://www.swiftbysundell.com/posts/picking-the-right-way-of-failing-in-swift